src/Player.h

```diff
@@ -1,4 +1,5 @@
 #pragma once
+#include "ResourceManager.h"
```

Comment:
we don't need to include that in .h
Instead, include that in .cpp files where it's actually used :)